anabot/runtime/installation/hub/software_selection.py

```diff
@@ -216,3 +220,52 @@ def addon_check(element, app_node, local_node):
     if action_result(element)[0] == False:
         return action_result(element)
     return addon_handler_manipulate(element, app_node, local_node, True)
+
+def get_full_kernel_option_name(kernel_option):
+    if kernel_option == '4k':
+        return FULL_4k_KERNEL_OPTION
+    elif kernel_option == '64k':
+        return FULL_64k_KERNEL_OPTION
+    else:
+        return None
+
+@handle_act('/kernel_options')
+def kernel_options_handler(element, app_node, local_node):
+    chosen_kernel_option = get_attr(element, "value")
+    if chosen_kernel_option is None:
+        return NotFoundResult("No kernel options specified", where="Anabot recipe.")
+    try:
+        kernel_combo_box = getnode(local_node, "combo box",
+                                   tr(DEFAULT_KERNEL_OPTION, context="GUI|Software Selection|Kernel Options"))
+    except TimeoutError:
+        return NotFoundResult(f"Couldn't find the {DEFAULT_KERNEL_OPTION} 'Kernel Options combo box'",
+                              where="Software Selection Spoke.")
+    # Open the combo box window by clicking on the combo box
+    kernel_combo_box.click()
+    try:
+        kernel_option_window = getnode(app_node, "window")
+    except TimeoutError:
+        return NotFoundResult("Couldn't find the opened 'Kernel Options' window", where="Software Selection Spoke.")
+    # The wanted kernel option node is specified by its full name, so we need to get it in order for anabot to find it
+    kernel_option_full_name = get_full_kernel_option_name(chosen_kernel_option)
+    try:
+        kernel_option_menu_item = getnode(kernel_option_window, "menu item",
+                                          tr(kernel_option_full_name, context="GUI|Software Selection|Kernel Options"))
+    except TimeoutError:
+        return NotFoundResult(chosen_kernel_option, where="Kernel Options combo box")
+    kernel_option_menu_item.click()
+    return ActionResultPass()
+
+@handle_chck('/kernel_options')
+def kernel_options_check(element, app_node, local_node):
+    if action_result(element)[0] == False:
```

Comment:
You can use [check_action_result decorator](https://github.com/rhinstaller/anabot/blob/main/anabot/runtime/decorators.py#L81) instead.